Misc/NEWS.d/next/Library/2024-11-08-17-05-10.gh-issue-120423.7rdLVV.rst

```diff
@@ -0,0 +1,2 @@
+Fix issue where :func:`urllib.request.pathname2url` mishandled Windows paths
+with embedded forward slashes.
```

Comment:
Backslashes that were also handled incorrectly:

- main:

```python
>>> from nturl2path import pathname2url
>>> pathname2url('//unc/as/path')
'//unc/as/path'
>>> pathname2url(r'\\unc\as\path')
'////unc/as/path'
```

- PR:

```python
>>> from nturl2path import pathname2url
>>> pathname2url('//unc/as/path')
'//unc/as/path'
>>> pathname2url(r'\\unc\as\path')
'//unc/as/path'
```

Reply:
Well, technically the original issue wasn't fixed then. I'll update the example.

Reply:
Sure. Your original repro case happened to be solved by #126208, but I took your bug report to be more general: that backward and forward slashes are handled differently. This patch should solve the last case where they're treated differently, even though it doesn't match your repro case.

Reply:
You can resolve this, the example now uses unc paths.